src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -968,34 +968,27 @@ private void UpdateWMStyles(Action change)
         {
             var oldDecorated = _decorated;
 
-            var oldThickness = BorderThickness;
-
             change();
 
             var style = (WindowStyles)GetWindowLong(_hwnd, (int)WindowLongParam.GWL_STYLE);
 
-            const WindowStyles controlledFlags = WindowStyles.WS_OVERLAPPEDWINDOW;
-
-            style = style | controlledFlags ^ controlledFlags;
-
             style |= WindowStyles.WS_OVERLAPPEDWINDOW;
 
             if (!_decorated)
             {
-                style ^= (WindowStyles.WS_CAPTION | WindowStyles.WS_SYSMENU);
+                style &= ~WindowStyles.WS_SYSMENU;
```

Comment:
>WS_CAPTION 0x00C00000L | The window has a title bar (includes the WS_BORDER style).

>WS_BORDER 0x00800000L | The window has a thin-line border.

I think this might add said "thin-line border". Have you checked with Win7/Win8/Win8.1?

Reply:
I haven't checked because I don't have a Win7/8 machine readily available. But I think it will also work on 7/8, because the `WM_NCCALCSIZE` message returns zero for undecorated windows. This is what removes the border:

https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-nccalcsize
> When wParam is TRUE, simply returning 0 without processing the NCCALCSIZE_PARAMS rectangles will cause the client area to resize to the size of the window, including the window frame. This will remove the window frame and caption items from your window, leaving only the client area displayed.

Reply:
Will check on Win 8.1 this week. I'm terribly sorry that I can't do that today or tomorrow, *extremely* busy.

Reply:
Found a Win7 machine, it works as expected:

![image](https://user-images.githubusercontent.com/43553916/68252673-6d41af80-0026-11ea-8325-20c0edf176e9.png)

Reply:
Also found a Win 8.1 machine, it also works as expected:

![image](https://user-images.githubusercontent.com/43553916/68341860-a4c56000-00e9-11ea-8e51-23ac0be89d4a.png)

Reply:
Thanks!